fgpyo/sam/__init__.py

```diff
@@ -863,40 +864,111 @@ def sum_of_base_qualities(rec: AlignedSegment, min_quality_score: int = 15) -> i
     return score
 
 
+def _set_common_mate_fields(dest: AlignedSegment, source: AlignedSegment) -> None:
+    """Set common mate info on a destination alignment from its mate's non-supplementary alignment.
```

Comment:
Having an offline discussion in Slack.  I think you want the mate's non-secondary and non-supplementary alignment.

Reply:
Yes! You are right. I'm never using a secondary alignment for `source` so all I need to do here is refine the documentation and add another assert to make sure no one sends in a secondary to `source`.